pkg/storageos/webhook.go

```diff
@@ -60,47 +65,60 @@ func (s Deployment) createWebhookService() error {
 // webhooks.  The configuration will be updated by the api-manager to include
 // the CA cert bundle.
 func (s Deployment) createMutatingWebhookConfiguration() error {
-	failurePolicy := admissionv1.Ignore
-	matchPolicy := admissionv1.Exact
-	sideEffects := admissionv1.SideEffectClassNoneOnDryRun
-	svcPort := webhookPort
 	scopeAll := admissionv1.AllScopes
-	path := "/mutate-pods"
-
 	webhooks := []admissionv1.MutatingWebhook{
-		{
-			Name: webhookPodMutatorName,
-			ClientConfig: admissionv1.WebhookClientConfig{
-				Service: &admissionv1.ServiceReference{
-					Name:      WebhookServiceName,
-					Namespace: s.stos.Spec.GetResourceNS(),
-					Port:      &svcPort,
-					Path:      &path,
+		s.mutatingWebhookConfiguration(webhookPodMutatorName, webhookPort, mutatePodPath, []admissionv1.RuleWithOperations{
+			{
+				Operations: []admissionv1.OperationType{admissionv1.Create},
+				Rule: admissionv1.Rule{
+					APIGroups:   []string{""},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"pods"},
+					Scope:       &scopeAll,
 				},
 			},
-			Rules: []admissionv1.RuleWithOperations{
-				{
-					Operations: []admissionv1.OperationType{admissionv1.Create},
-					Rule: admissionv1.Rule{
-						APIGroups:   []string{""},
-						APIVersions: []string{"v1"},
-						Resources:   []string{"pods"},
-						Scope:       &scopeAll,
-					},
+		}),
+		s.mutatingWebhookConfiguration(webhookPVCMutatorName, webhookPort, mutatePVCPath, []admissionv1.RuleWithOperations{
+			{
+				Operations: []admissionv1.OperationType{admissionv1.Create},
+				Rule: admissionv1.Rule{
+					APIGroups:   []string{""},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"persistentvolumeclaims"},
+					Scope:       &scopeAll,
 				},
 			},
-			FailurePolicy:           &failurePolicy,
-			MatchPolicy:             &matchPolicy,
-			NamespaceSelector:       &metav1.LabelSelector{},
-			SideEffects:             &sideEffects,
-			AdmissionReviewVersions: []string{"v1"},
-		},
+		}),
 	}
 	labels := podLabelsForAPIManager(s.stos.Name)
 
 	return s.k8sResourceManager.MutatingWebhookConfiguration(MutatingWebhookConfigName, labels, webhooks).Create()
 }
 
+func (s Deployment) mutatingWebhookConfiguration(name string, port int32, path string, rules []admissionv1.RuleWithOperations) admissionv1.MutatingWebhook {
+	failurePolicy := admissionv1.Ignore
```

Comment:
Shouldn't this also be a function parameter to be able to have different failure policies for Pod and PVC webhook configurations?

Reply:
Yes, I've changed it to be a function param but left the PVC webhook to ignore failures for now.  I want to do more testing before setting to fail.